src/Swift.Bindings/src/Model/SwiftFunction.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents a Swift function signature and its provenance.
+/// </summary>
+[DebuggerDisplay("{ToString()}")]
+public class SwiftFunction {
```

Comment:
Why can't we utilize MethodDecl and it's members instead?

Reply:
See above. This is strictly Swift not C#.